mod/http_repo_test.go

```diff
@@ -0,0 +1,41 @@
+package mod
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestDownload(t *testing.T) {
+	cases := []struct {
+		name    string
+		require Require
+		auth    string
+	}{
+		{
+			name: "download archives",
+			require: Require{
+				cloneRepo: "https://github.com/dagger/dagger/archive/refs/tags/v0.2.7.tar.gz",
+				clonePath: "",
+				version:   "main",
+			},
+		},
+	}
+
+	for _, c := range cases {
```

Comment:
Wouldn't it be better to use https://pkg.go.dev/net/http/httptest and make this a unit test?